docs/program/connectivity.md

```diff
@@ -3,26 +3,24 @@ title: "Client Sessions and Robot Network Connectivity"
 linkTitle: "Network Connectivity Issues"
 weight: 20
 type: "docs"
-description: "Whan a robot loses its connection to the internet, all client sessions will timeout and end by default."
+description: "When a robot loses its connection to a LAN or WAN, all client sessions will timeout and end by default."
 tags:
   ["client", "sdk", "viam-server", "networking", "apis", "robot api", "session"]
 ---
 
-What happens when your robot loses its connection to the internet over a LAN or WAN and can no longer communicate with clients through [the Viam app](https://app.viam.com).
+When connecting to a robot using the connection code from the [code sample tab](/program/#hello-world-the-code-sample-tab), a [client session](/program/apis/sessions/) automatically uses the most efficient route to connect to your robot either through local LAN or WAN or the internet.
 
-## How does `viam-server` work if your robot loses its network connection?
+When a robot loses its connection to the internet but is still connected to a LAN or WAN:
 
-When a robot loses its connection over LAN or WAN, it can no longer communicate with clients through [the Viam app](https://app.viam.com).
-When no heartbeat has been received over a timeout period, `viam-server` will end any current client [_sessions_](/program/apis/sessions/) on this robot.
+- Client sessions connected through the same LAN or WAN will function normally.
+- Client sessions connected through the internet will timeout and end.
+  If the client is on the same LAN or WAN, the client will automatically disconnect and then reconnect over LAN.
```

Comment:
this seems to claim we will see a disconnect when this occurs, but i would expect that only happens if the connection was for some reason not direct to bot. in the case where it was not direct, (based on some of cheuks claims), i would expect it cannot connect locally? 

perhpas my expectation of non local connect is wrong, in which case, i'd still expected it wouldnt only disconnect in some cases (those where it was connected via internet/external connections)

Reply:
@cheukt should I remove the last sentence since very likely sessions would already have chosen the most efficient route through LAN/WAN?

Reply:
did some manual testing, I would say it's very likely that sessions (which usually just uses the client connection) would have chosen the most efficient route, which would be direct to robot over the LAN/WAN

Reply:
so last line is probably not needed

Reply:
How about this:
```suggestion
  If the client is on the same LAN or WAN but the route it chose to connect is through the internet, the client will automatically disconnect and then reconnect over LAN.
```